### ISSUE TITLE
Improve highlighting

### ISSUE THREAD

**Issue by sigmaSd** [sigmaSd/IRust#55]
Currently highlighting works pretty well, but one issue remains if we paste code into the repl it becomes noticeably laggy, I'm not sure whats the best way to deal with it

1 - we can de-bounce from highlighting calls, that is if the user input very fast(like pasting) write all the input without highlighting and then highlight when the input finishes

2 - Investigate how editors (exp: [accepted](https://github.com/hatoo/accepted)) that use syntect do it, I'm only using the easy module, there is probably a more correct way to highlight text

**Comment by sigmaSd** [comment]
1- is actually possible see https://github.com/sigmaSd/IRust/tree/async_highlight but crossterm async story is not good for now, (its a busy loop that will consume a lot of the cpu), maybe revisit this after its reworked upstream, or figure how to handle it correctly with the current implementation

**Comment by sigmaSd** [comment]
Fixed in https://github.com/sigmaSd/IRust/commit/0a0e9fc0370ab5e5214d35e25315dc667962f11a
tldr: Removed syntect, used a custom implemented parser
